tables/DataMan/test/tDataManInfo.cc

```diff
@@ -0,0 +1,230 @@
+#include <casacore/casa/iostream.h>
+
+using namespace casacore;
+using namespace std;
+
+// <summary>
+// Test program for class DataManInfo
+// </summary>
+
+// Test modifying the dminfo record.
+void testDM()
+{
+    TableDesc td;
+    td.addColumn(ScalarColumnDesc<Int>("col1"));
+    td.addColumn(ScalarColumnDesc<Int>("col2"));
+    td.addColumn(ScalarColumnDesc<Int>("col3"));
+    td.addColumn(ArrayColumnDesc<Int>("col4"));
+    // Now create a new table from the description.
+    SetupNewTable aNewTab("tTableCopy_tmp.dm", td, Table::New);
+    Table tabl(aNewTab);
+    Record dminfo = tabl.dataManagerInfo();
+    cout << dminfo;
+    Vector<String> remCols1 =
+      DataManInfo::removeDminfoColumns (dminfo, Vector<String>(1, "col1"), "Standard");
+    cout << dminfo << remCols1 << endl;
+    Vector<String> remCols2 =
+      DataManInfo::removeDminfoColumns (dminfo, Vector<String>(1, "col1"));
+    cout << dminfo << remCols2 << endl;
+    DataManInfo::setTiledStMan (dminfo, Vector<String>(1, "col3"),
+                                "TiledShapeStMan", "TSMData", IPosition(3,3,4,5));
+    cout << dminfo << endl;
+}
+
+void mergeTestEmpty()
+{
+  Record rec;
+  DataManInfo::mergeInfo (rec, Record());
+  AlwaysAssertExit (rec.nfields() == 0);
+}
+
+void mergeTestNonEmpty()
+{
+  Record dminfo;
+  Record sub;
+  sub.define("TYPE", "abc");
+  sub.define ("COLUMNS", Vector<String>(1, "col1"));
+  dminfo.defineRecord (0, sub);
+  Record rec;
+  DataManInfo::mergeInfo (rec, dminfo);
+  AlwaysAssertExit (rec.nfields() == 1);
+  const Record& dm1 = rec.subRecord(0);
+  AlwaysAssertExit (dm1.asString("TYPE") =="abc");
+  AlwaysAssertExit (! dm1.isDefined("NAME"));
+  AlwaysAssertExit (dm1.asArrayString("COLUMNS").size() == 1);
+  AlwaysAssertExit (dm1.asArrayString("COLUMNS").data()[0] == "col1");
+}
+
+void finalizeTestEmpty()
+{
+  Record rec = DataManInfo::finalizeMerge (TableDesc(), Record());
+  AlwaysAssertExit (rec.nfields() == 0);
+}
+
+void finalizeTestNonEmpty()
+{
+  Record dminfo;
+  Record sub;
+  sub.define("TYPE", "abc");
+  sub.define("NAME", String());
+  sub.define ("COLUMNS", Vector<String>(1, "col1"));
+  dminfo.defineRecord (0, sub);
+  {
+    Record rec = DataManInfo::finalizeMerge (TableDesc(), dminfo);
+    AlwaysAssertExit (rec.nfields() == 0);
+  }
+  TableDesc desc;
+  desc.addColumn (ScalarColumnDesc<Int>("col1"));
+  Record rec;
+  DataManInfo::mergeInfo (rec, dminfo);
+  rec = DataManInfo::finalizeMerge (desc, dminfo);
+  AlwaysAssertExit (rec.nfields() == 1);
+  const Record& dm1 = rec.subRecord(0);
+  AlwaysAssertExit (dm1.nfields() == 3);
+  AlwaysAssertExit (dm1.asString("TYPE") =="abc");
+  AlwaysAssertExit (dm1.asString("NAME") == "col1");
+  AlwaysAssertExit (dm1.asArrayString("COLUMNS").size() == 1);
+  AlwaysAssertExit (dm1.asArrayString("COLUMNS").data()[0] == "col1");
+}
+
+void largeTest()
+{
+  // Create a table description with various data manager types and names.
+  TableDesc desc;
```

Comment:
`desc` is only defined but not used in this test...

Reply:
Forgot to do the finalizeMerge, which I've added now.